### PR TITLE
drivers: model IEEE 802.15.4 devices in Kconfig

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This models all IEEE 802.15.4 device drivers and the related modules (mostly netdev). Similarly to #17739, this modifies the related tests to avoid using the full GNRC stack. Some tests allow access to device-specific features (e.g., batmon in at86rf215), all these are kept.

A question regarding modelling + UI (maybe @jia200x has some insight): currently a radio can implement netdev or the radio HAL. In the second case, when the top level user (e.g. a network stack or the current tests) requires to interact via the netdev interface, the `netdev_ieee802154_submac` module is required to fill the gap. So far, I added a symbol that indicates the presence of a radio HAL implementer (selected by the specific driver). Depending on whether `netdev` is used, the submac module is used (if `netdev` is used, and a driver implements radio HAL -> bring submac). This has a shortcoming: if an application (or other module) wants to use two devices implementing different interfaces (netdev vs. HAL), the submac will be pulled. We could create an extra symbol (e.g., `USE_NETDEV_INTERFACE`) that indicates that whatever network device you have, you want to interact using netdev, but I'm not sure that's the way. Thoughts?

### Testing procedure
- Green CI

### Issues/PRs references
~~Depends on #17790~~
~~Depends on #17791~~
~~Depends on #17792~~

### CODE REVIEW

File: drivers/at86rf215/Kconfig [hunk 1]
```diff
@@ -196,27 +196,181 @@ config AT86RF215_DEFAULT_MR_FSK_MOD_IDX
...
+config MODULE_AT86RF215_TIMESTAMP
+    bool "Enable timestamp frame information"
+
+config MODULE_AT86RF215_BLOCKING_SEND
+    bool "Block while sending"
 
-endmenu
+config MODULE_AT86RF215_SUBGHZ
+    bool
+    prompt "Sub-GHz support" if !HAVE_AT86RF215M
```
Comment (benpicco) [inline drivers/at86rf215/Kconfig:331]: That has it reversed - AT86RF215M is sub-GHz only

Comment (leandrolanzieri) [reply drivers/at86rf215/Kconfig:331]: Yeah, that is what I tried modelling here. When `HAVE_AT86RF215M` is there the prompt is not visible and defaults to `y` (i.e., the user can't de-select the sub-GHz support, which is enabled).

**Comment by leandrolanzieri** [comment]
I split some dependencies

**Comment by leandrolanzieri** [comment]
Rebased again

File: drivers/kw2xrf/Kconfig [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2022 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+menuconfig MODULE_KW2XRF
+    bool "KW2XRF radio"
```
Comment (MrKevinWeiss) [inline drivers/kw2xrf/Kconfig:9]: Maybe you should be using the:
```kconfig
    bool
    prompt "KW2XRF radio" if !(MODULE_NETDEV_DEFAULT && HAVE_KW2XRF)
    default (MODULE_NETDEV_DEFAULT && HAVE_KW2XRF)
```

as this is a module with defaults and options?

**Review by MrKevinWeiss — COMMENTED** [review]
Small thing, not sure if it matters.  Lets see what murdock says!

File: drivers/kw2xrf/Kconfig (hunk 2 above)
Comment (leandrolanzieri) [reply drivers/kw2xrf/Kconfig]: Sure, fixed and removed the `select` from the feature

**Comment by MrKevinWeiss** [comment]
Needs a rebase after the merge as well...

**Comment by leandrolanzieri** [comment]
Looking good @MrKevinWeiss

**Comment by MrKevinWeiss** [comment]
Can we rerun 1 more time now that the  kconfig checks are actually working?

**Review by MrKevinWeiss — APPROVED** [review]
I will pre ack though!

**Comment by MrKevinWeiss** [comment]
And [that](https://ci.riot-os.org/RIOT-OS/RIOT/17789/1b21ee38c56cb0012e3f99a2b9b63c8e99adc6a9/output/compile/tests/ieee802154_hal/remote-revb:gnu.txt) is why we check 😄

**Pull request merged**